### PR TITLE
Weekly Project 1

### PR DESCRIPTION
**Pull request by PurpleDragon1999**
Profile page and Weekly Project 1(Tic-Tac-Toe).

### CODE REVIEW

File: Shubham_Sharma/Profile-bootstrap/Profile.html [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+                    <footer class="blockquote-footer"><b>On my way to <cite title="Source Title">Become Professional</cite></b></footer>
+                </div>
+          </div>
+        </div>
+        <div class="col-sm-1">
+
+        </div>
+    </div>
+    <div class="row mb-5">
+        <div class="col-sm-3">
```
Comment (nikhilrstg18) [inline Shubham_Sharma/Profile-bootstrap/Profile.html]: why empty div ?

File: Shubham_Sharma/Profile-bootstrap/Profile.html [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+        <div class="col-sm-5 mb-3">
+            <div class="card bg-pic">
+                <div class="card-block text-center py-3">
+                    <h3 class="card-title">CSS</h3>
+                    <p class="card-text"><b>Intermediate</b></p>
+                    <footer class="blockquote-footer"><b>On my way to <cite title="Source Title">Become Professional</cite></b></footer>
+                </div>
+          </div>
+        </div>
+        <div class="col-sm-1">
```
Comment (nikhilrstg18) [inline Shubham_Sharma/Profile-bootstrap/Profile.html]: why empty div

File: Shubham_Sharma/Profile-bootstrap/Profile.html [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+
+        </div>
+    </div>
+
+    <div class="row mb-5">
+        <div class="col-sm-3">
```
Comment (nikhilrstg18) [inline Shubham_Sharma/Profile-bootstrap/Profile.html]: why empty div

File: Shubham_Sharma/Tic-Tac-Toe/Assets/js/Main.js [hunk 4]
```diff
@@ -0,0 +1,167 @@
+var options = "X";
+
+//Resets every button's text and color
+function reset(){
+    var b1 = document.getElementById("b1");
```
Comment (nikhilrstg18) [inline Shubham_Sharma/Tic-Tac-Toe/Assets/js/Main.js]: missing DRY

File: Shubham_Sharma/Tic-Tac-Toe/Assets/js/Main.js [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+    var b3 = document.getElementById("b3").innerHTML;
+    var b4 = document.getElementById("b4").innerHTML;
+    var b5 = document.getElementById("b5").innerHTML;
+    var b6 = document.getElementById("b6").innerHTML;
+    var b7 = document.getElementById("b7").innerHTML;
+    var b8 = document.getElementById("b8").innerHTML;
+    var b9 = document.getElementById("b9").innerHTML;
+    
+
+    if(((b1=="X") || (b1=="O")) && ((b1 == b2) && (b2 == b3))){
```
Comment (nikhilrstg18) [inline Shubham_Sharma/Tic-Tac-Toe/Assets/js/Main.js]: get rid if if-else-if ladder and missing DRY

File: Shubham_Sharma/Tic-Tac-Toe/Assets/js/Main.js [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+function highLight(cell1, cell2, cell3, symbol){
+    btnDisable();
+    document.getElementById(cell1).style.backgroundColor = '#FFA500';
+    document.getElementById(cell2).style.backgroundColor = '#FFA500';
+    document.getElementById(cell3).style.backgroundColor = '#FFA500';
+}
+
+//The button pressed will be disabled and symbol 'X' or 'O' will appear on it
+function setButton(button, options){
+    if(button == 1){
```
Comment (nikhilrstg18) [inline Shubham_Sharma/Tic-Tac-Toe/Assets/js/Main.js]: missing DRY, code here can be reduced to 4 to 5 lines of code only

File: Shubham_Sharma/Tic-Tac-Toe/Assets/js/Main.js [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+    checker();
+}
+
+//Checks which symbol is next going to appear, like if last symbol was X, then O will be printed next 
+function clicker(button){
+    if(options == "X"){
+        options = "O";
+        setButton(button, options);
+    }
+    else if(options == "O"){
```
Comment (nikhilrstg18) [inline Shubham_Sharma/Tic-Tac-Toe/Assets/js/Main.js]: is there a need to use else here?

**Review by nikhilrstg18 — CHANGES_REQUESTED** [review]
MISSING DRY KISS
extra code and code optimization has great scope
Good to see step commenting 
by the way, I see there are 62 files, are these all utilized ??

File: Shubham_Sharma/Profile-bootstrap/Profile.html (hunk 1 above)
Comment (PurpleDragon1999) [reply Shubham_Sharma/Profile-bootstrap/Profile.html]: I am using bootstrap grid, so to center the elements, I am using this empty div and setting columns.

File: Shubham_Sharma/Profile-bootstrap/Profile.html (hunk 2 above)
Comment (PurpleDragon1999) [reply Shubham_Sharma/Profile-bootstrap/Profile.html]: I am using bootstrap grid, so to center the elements, I am using this empty div and setting columns.

File: Shubham_Sharma/Profile-bootstrap/Profile.html (hunk 3 above)
Comment (PurpleDragon1999) [reply Shubham_Sharma/Profile-bootstrap/Profile.html]: I am using bootstrap grid, so to center the elements, I am using this empty div and setting columns.

File: Shubham_Sharma/Tic-Tac-Toe/Assets/js/Main.js (hunk 5 above)
Comment (PurpleDragon1999) [reply Shubham_Sharma/Tic-Tac-Toe/Assets/js/Main.js]: Can I use ternary operator instead ?

File: Shubham_Sharma/Tic-Tac-Toe/Assets/js/Main.js (hunk 6 above)
Comment (PurpleDragon1999) [reply Shubham_Sharma/Tic-Tac-Toe/Assets/js/Main.js]: Okay, resolved that.

File: Shubham_Sharma/Tic-Tac-Toe/Assets/js/Main.js [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+    var b2 = document.getElementById("b2").innerHTML;
+    var b3 = document.getElementById("b3").innerHTML;
+    var b4 = document.getElementById("b4").innerHTML;
+    var b5 = document.getElementById("b5").innerHTML;
+    var b6 = document.getElementById("b6").innerHTML;
+    var b7 = document.getElementById("b7").innerHTML;
+    var b8 = document.getElementById("b8").innerHTML;
+    var b9 = document.getElementById("b9").innerHTML;
+    
+    (((b1=="X") || (b1=="O")) && ((b1 == b2) && (b2 == b3))) ? highLight("b1", "b2", "b3", b1)
```
Comment (nikhilrstg18) [inline Shubham_Sharma/Tic-Tac-Toe/Assets/js/Main.js]: still violation of DRY

suggestion: ternary operator should be used up to 3 levels is good if more than max 3 levels, you should change your approach

**Review by nikhilrstg18 — CHANGES_REQUESTED** [review]
@PurpleDragon1999 
please reply on each comment about the resolution of it before marking in resolved

File: Shubham_Sharma/Tic-Tac-Toe/Assets/js/Main.js (hunk 8 above)
Comment (PurpleDragon1999) [reply Shubham_Sharma/Tic-Tac-Toe/Assets/js/Main.js]: Resolved the violation of DRY. Kindly check and let me know if anything else is there.

**Review by nikhilrstg18 — APPROVED** [review]
Looks good to me

**Review by amandeeprikhi — APPROVED** [review]
LGTM

**Comment by vishivish18** [comment]
@PurpleDragon1999  Please fix the conflicts and ask for as merge

**Pull request merged**